**Notas para contenedor de docker:**

Comando de docker para ejecución de la nota de forma local:

nota: cambiar `dir_montar` por la ruta de directorio que se desea mapear a `/datos` dentro del contenedor de docker.

```
dir_montar=<ruta completa de mi máquina a mi directorio>#aquí colocar la ruta al directorio a montar, por ejemplo: 
#dir_montar=/Users/erick/midirectorio.
```

Ejecutar:

```
$docker run --rm -v $dir_montar:/datos --name jupyterlab_prope_r_kernel_tidyverse -p 8888:8888 -d palmoreck/jupyterlab_prope_r_kernel_tidyverse:2.1.4   

```

Ir a `localhost:8888` y escribir el password para jupyterlab: `qwerty`

Detener el contenedor de docker:

```
docker stop jupyterlab_prope_r_kernel_tidyverse
```


Documentación de la imagen de docker `palmoreck/jupyterlab_prope_r_kernel_tidyverse:2.1.4` en [liga](https://github.com/palmoreck/dockerfiles/tree/master/jupyterlab/prope_r_kernel_tidyverse).

---

Para ejecución de la nota usar:

[docker](https://www.docker.com/) (instalación de forma **local** con [Get docker](https://docs.docker.com/install/)) y ejecutar comandos que están al inicio de la nota de forma **local**. 

O bien dar click en alguno de los botones siguientes:

[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/palmoreck/dockerfiles-for-binder/jupyterlab_prope_r_kernel_tidyerse?urlpath=lab/tree/Propedeutico/R/clases/2_probabilidad/5_independencia.ipynb) esta opción crea una máquina individual en un servidor de Google, clona el repositorio y permite la ejecución de los notebooks de jupyter.

[![Run on Repl.it](https://repl.it/badge/github/palmoreck/dummy)](https://repl.it/languages/Rlang) esta opción no clona el repositorio, no ejecuta los notebooks de jupyter pero permite ejecución de instrucciones de Python de forma colaborativa con [repl.it](https://repl.it/). Al dar click se crearán nuevos ***repl*** debajo de sus users de ***repl.it***.


### Lo siguiente está basado los libros:

### * B. Rumbos, Pensando Antes de Actuar: Fundamentos de Elección Racional, 2009.

### * G. J. Kerns, Introduction to Probability and Statistics Using R, 2014.

### El libro de G. J. Kerns tiene github: [jkerns/IPSUR](https://github.com/gjkerns/IPSUR)

**Notas:**

* Se utilizará el paquete [prob](https://cran.r-project.org/web/packages/prob/index.html) de R para los experimentos descritos en la nota y aunque con funciones nativas de R se pueden crear los experimentos, se le da preferencia a mostrar cómo en R se tienen paquetes para muchas aplicaciones.

* En algunas líneas no es necesario colocar `print` y sólo se ha realizado para mostrar los resultados de las funciones en un formato similar al de R pues la nota se escribió con jupyterlab y R.

* Cuidado al utilizar las funciones del paquete `prob` para construir espacios de probabilidad grandes como lanzar un dado 9 veces... (tal experimento tiene 10 millones de posibles resultados)

In [1]:
library(prob)

Loading required package: combinat


Attaching package: ‘combinat’


The following object is masked from ‘package:utils’:

    combn


Loading required package: fAsianOptions

Loading required package: timeDate

Loading required package: timeSeries

Loading required package: fBasics

Loading required package: fOptions


Attaching package: ‘prob’


The following objects are masked from ‘package:base’:

    intersect, setdiff, union




# Eventos independientes

Dos eventos son independientes si la ocurrencia de uno de ellos es irrelevante para el otro.

## Ejemplo

**Experimento:** lanzar dos monedas **distintas** (o una moneda dos veces). El espacio de resultados es: $S=\{AA, AS, SA, SS\}$. Sean los eventos $E=${primer lanzamiento es A} y $F=${segundo lanzamiento es A}. Entonces $P(E)=\frac{2}{4}$, $P(F)=\frac{2}{4}$ y $P(E \cap F)=\frac{1}{4}$. 

Se tiene: $$P(F| E) = \frac{P(E \cap F)}{P(E)} = \frac{P({\{AA\}})}{P(E)} = \frac{\frac{1}{4}}{\frac{2}{4}} = \frac{1}{2} = P(F)$$

La igualdad anterior implica que la información de que el primer lanzamiento es $A$ no afecta o modifica la probabilidad que el segundo lanzamiento sea $A$.

Los eventos $E$ y $F$ son independientes si $P(E \cap F) = P(E)P(F)$ en otro caso se les nombra dependientes.

**Obs:** Obsérvese que si $P(F) > 0$ entonces se puede definir la independencia de eventos como: $E$ y $F$ son independientes si $P(E|F) = \frac{P(E \cap F)}{P(F)} = \frac{P(E)P(F)}{P(F)} = P(E)$.

## Propiedades

Si $E$ y $F$ son independientes entonces:

* $E$ y $F^c$ son independientes,

* $E^c$ y $F$ son independientes,

* $E^c$ y $F^c$ son independientes.

**Obs:** si los eventos $E$ y $F$ son mutuamente excluyentes (no ocurren simultáneamente, también nombrados ajenos) y tienen probabilidades de ocurrencia positiva entonces no pueden ser independientes pues $P(E \cap F) = 0$ (por mutuamente excluyentes) y $P(E)P(F)>0$ (por probabilidad positiva de ambos) por lo que $P(E \cap F) \neq P(E)P(F)$. Lo anterior es equivalente a: si $E$ y $F$ son independientes con probabilidad positiva entonces no pueden ser mutuamente excluyentes (ocurren simultáneamente).

**Obs2:** Para tres eventos $E$, $F$ y $G$, se escribe que son independientes si:

1) $P(E \cap F) = P(E)P(F)$

2) $P(E \cap G) = P(E)P(G)$

3) $P(F \cap G) = P(F)P(G)$

4) $P(E \cap F \cap G) = P(E)P(F)P(G)$.

Si sólo se cumplen $1,2,3$ entonces se llaman independientes por pares.

## Ejemplo

1) **Experimento:** lanzar $10$ monedas. Calcular la probabilidad de que en al menos una moneda salga águila y supóngase que los lanzamientos no interfieren con los demás.

**Solución:** Sean $E_i$ los eventos {sale águila en la moneda $i$} para $i=1,\dots,10$. Entonces son eventos independientes por suposición. Si $F$ es el evento {al menos una moneda sale águila} se tiene: $P(F) = 1-P(F^c)$ con $F^c$ el evento {no salen águilas} = {salen $10$ soles}. Entonces:

$$P(F) = 1-P(F^c) = 1-P(E_1^c \cap E_2^c \dots \cap E_{10}^c) = 1- P(E_1^c)P(E_2^c) \cdots P(E_{10}^c) = 1-\left(\frac{1}{2}\right)^{10} \approx 0.9990234$$.

### En R ...

In [5]:
S<-tosscoin(10, makespace=T) # T = True

In [6]:
head(S)

toss1 toss2 toss3 toss4 toss5 toss6 toss7 toss8 toss9 toss10 probs       
1 H     H     H     H     H     H     H     H     H     H      0.0009765625
2 T     H     H     H     H     H     H     H     H     H      0.0009765625
3 H     T     H     H     H     H     H     H     H     H      0.0009765625
4 T     T     H     H     H     H     H     H     H     H      0.0009765625
5 H     H     T     H     H     H     H     H     H     H      0.0009765625
6 T     H     T     H     H     H     H     H     H     H      0.0009765625

In [8]:
E<-subset(S, isrep(S, vals='T',nrep = 10)) 
#equivalently: E2<-subset(S,isin(S, rep('T',10)))
E

toss1 toss2 toss3 toss4 toss5 toss6 toss7 toss8 toss9 toss10 probs       
1024 T     T     T     T     T     T     T     T     T     T      0.0009765625

In [9]:
1-Prob(E)

[1] 0.9990234

## Independencia condicional

El concepto de independencia puede extenderse a independencia condicional; concretamente, tenemos la siguiente definición: Sean $E, F$ y $G$ eventos con probabilidades positivas. Entonces, los eventos $E$ y $F$ son condicionalmente independientes dado $G$ si y sólo si, $P(E \cap F|G) = P(E|G)P(F|G)$, esto es, $E$ dado $G$ y $F$ dado $G$ son eventos independientes.

## Experimentos repetidos que cumplen independencia y condiciones idénticas en R

Es muy común en la práctica realizar un experimento múltiples veces bajo condiciones idénticas y de una forma independiente, por ejemplo: lanzamiento de una moneda o de un dado.

En *R* la función [iidspace](https://www.rdocumentation.org/packages/prob/versions/1.0-1/topics/iidspace) del paquete `prob` fue diseñada para lo anterior. Tiene tres argumentos: `x` que es un vector de *outcomes*, `ntrials` que es un entero que indica el número de veces que se repite el experimento y `probs` que especifican las probabilidades de los `outcomes` en `x` en una sola ejecución.

### Ejemplo

**Experimento:** Tres lanzamientos de una moneda cargada con probabilidad de obtener águila de $0.7$ y sol de $0.3$.

In [10]:
iidspace(c('aguila','sol'), ntrials = 3, probs = c(0.7, 0.3))

X1     X2     X3     probs
1 aguila aguila aguila 0.343
2 sol    aguila aguila 0.147
3 aguila sol    aguila 0.147
4 sol    sol    aguila 0.063
5 aguila aguila sol    0.147
6 sol    aguila sol    0.063
7 aguila sol    sol    0.063
8 sol    sol    sol    0.027

Obsérvese que: $P({AAA}) = .7*.7*.7 = 0.343$ y el resultado de la función es un espacio de probabilidad.

El procedimiento anterior puede usarse para modelar un dado cargado o cualquier otro experimento que pueda ser representado con un vector de posibles `outcomes`. 

## Ejercicios

**Nota: Varios de los siguientes ejercicios tienen su solución en las referencias dadas al inicio de la nota. Se sugiere resolverlos por su cuenta y posteriormente compararlos con sus respectivas soluciones.**

**1)** Calcular la probabilidad de sacar un as y corazón en una baraja

In [3]:
(4/52)*(13/52)

[1] 0.01923077

In [4]:
1/52

[1] 0.01923077

**2)** Se lanzan dos dados, uno verde y otro rojo. Considérense los eventos $E=${los números suman 5}, $F=${el número en el dado verde es impar}. ¿$E$ y $F$ son independientes?.

$P(E)$

In [39]:
S_E <- rolldie(2, 6, makespace=T)
S_E$Sum <- S_E$X1 + S_E$X2
S_E <- S_E[S_E$Sum == 5, ]

In [40]:
S_E

X1 X2 probs      Sum
4  4  1  0.02777778 5  
9  3  2  0.02777778 5  
14 2  3  0.02777778 5  
19 1  4  0.02777778 5

In [47]:
P_E <- sum(S_E$probs)
P_E

[1] 0.1111111

$P(F)$

In [80]:
S_F <- rolldie(1, 6, makespace=T)
S_F <- S_F[S_F$X1 %in% c(1, 3, 5), ]

In [81]:
S_F

X1 probs    
1 1  0.1666667
3 3  0.1666667
5 5  0.1666667

In [82]:
P_F <- sum(S_F$probs)
P_F

[1] 0.5

$P(E \cap F)$

In [97]:
S_EyF <- rolldie(2, 6, makespace=T)
S_EyF$Sum <- S_EyF$X1 + S_EyF$X2
S_EyF <- S_EyF[(S_EyF$X2 %in% c(1, 3, 5)) & (S_EyF$Sum == 5), ]

In [98]:
S_EyF

X1 X2 probs      Sum
4  4  1  0.02777778 5  
14 2  3  0.02777778 5

In [99]:
P_EyF <- sum(S_EyF$probs)
P_EyF

[1] 0.05555556

Validación de:
$$P(E \cap F) = P(E)P(F)$$

In [100]:
P_E * P_F == P_EyF

[1] TRUE

Por lo tanto, los dos eventos sí son independientes

**3)** Supóngase que nuestra elección de postre cuando salimos a comer es independiente del platillo principal que pedimos. La mitad de las veces que vamos a comer al restaurante "L Intervallo" pedimos pasta con camarones y la mitad de las veces pedimos *panna* *cotta* de postre. ¿Cuál es la probabilidad de ordenar pasta con camarones y *panna* *cotta* dado que comemos en "L Intervallo"?

Probabilidad de pedir pasta con camarones $ \Rightarrow P(P) = 0.5$

Probabilidad de pedir *panna cotta* $ \Rightarrow P(C) = 0.5$

Por lo tanto:

**Probabilidad de pedir pasta con camarones y *panna cotta* $ \Rightarrow P(P \cap C) = \frac{1}{2}\frac{1}{2} = \frac{1}{4} $**

**4)** Sea E el evento de que el número $111111$ sea ganador en el sorteo Melate (la lotería Melate consiste en elegir una combinación de seis números diferentes entre el $1$ y el $56$, sin orden y sin reemplazo) y $F$ el evento de que el número $555555$ sea el ganador.

a) Determina si estos eventos son mutuamente excluyentes.

**Sí son mutuamente excluyentes dado que no puede salir el número ganador 111111 y 555555 al mismo tiempo**

b) Determina si estos eventos son independientes.

**Dado que se trata de eventos mutuamente excluyentes, se trata también de eventos que no pueden ser independientes**